contribs/gmf/src/directives/partials/layertree.html

```diff
@@ -0,0 +1,17 @@
+<span ng-if="::!layertreeCtrl.isRoot"
+    ng-class="['depth-' + layertreeCtrl.depth, gmfLayertreeCtrl.getResolutionStyle(layertreeCtrl.node), gmfLayertreeCtrl.getNoSourceStyle(layertreeCtrl)]">
+  <a href ng-click="::gmfLayertreeCtrl.toggleActive(layertreeCtrl)">
+    <span class="state" ng-class="[gmfLayertreeCtrl.getSetActive(layertreeCtrl)]"></span>
```

Comment:
you could do a `ng-if` here with one time binding to know if there are children.
Only use the `ng-class` when children will avoid you a lot of useless watch functions executions.